src/target/riscv/batch.c

```diff
@@ -8,6 +8,7 @@
 
 #define get_field(reg, mask) (((reg) & (mask)) / ((mask) & ~((mask) << 1)))
 #define set_field(reg, mask, val) (((reg) & ~(mask)) | (((val) * ((mask) & ~((mask) << 1))) & (mask)))
+#define DIM(x)		(sizeof(x)/sizeof(*x))
```

Comment:
I recently discovered that OpenOCD already defines ARRAY_SIZE that does exactly this in src/helper/types.h, which is already included just about everywhere.

Reply:
Ah OK, ARRAY_SIZE it is (probably not a bad idea to familiarize myself with other utility macros and functions)